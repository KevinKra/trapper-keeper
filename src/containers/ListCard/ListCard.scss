@import "../../variables.scss";

.ListCard {
  @include container-styling(300px);
  box-shadow: none;
  &:hover {
    cursor: pointer;
    box-shadow: 1px 1px 5px grey;
  }
  ul {
    list-style-type: none;
    margin: auto;
    padding: 1rem;
    div {
      display: flex;
    }
  }
}
