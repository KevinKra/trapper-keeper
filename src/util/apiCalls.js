export const getAllLists = async () => {
  const res = await fetch(`${process.env.REACT_APP_BASE_URL}/api/v1/lists`);
  if (!res.ok) {
    throw new Error("Could not get lists");
  }
  return await res.json();
};

export const addNewList = async list => {
  const res = await fetch(`${process.env.REACT_APP_BASE_URL}/api/v1/lists`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(list)
  });
  if (!res.ok) {
    throw new Error("Could not save new list");
  }
  return await res.json();
};

export const getListData = async id => {
  const res = await fetch(
    `${process.env.REACT_APP_BASE_URL}/api/v1/lists/${id}`
  );
  if (!res.ok) {
    throw new Error("Could not fetch list with that id");
  }
  return await res.json();
};

export const deleteList = async id => {
  const res = await fetch(
    `${process.env.REACT_APP_BASE_URL}/api/v1/lists/${id}`,
    {
      method: "DELETE"
    }
  );
  if (!res.ok) {
    throw new Error("Could not delete card");
  }
  return await res.json();
};
