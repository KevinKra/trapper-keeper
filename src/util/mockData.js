export const mockNote = {
	id: Date.now(),
	userTask: 'Hello, world!',
	complete: false
};

export const mockList = {
	title: 'My List',
	notes: [mockNote, mockNote, mockNote]
};

export const mockLists = [mockList, mockList, mockList];
