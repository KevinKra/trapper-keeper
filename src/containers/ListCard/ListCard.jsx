import React, { Component } from 'react';
import ListItem from '../../components/ListItem/ListItem';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import * as actions from '../../actions';
import * as apiCalls from '../../util/apiCalls';

import './ListCard.scss';

export class ListCard extends Component {
	deleteList = async () => {
		try {
			await apiCalls.deleteList(this.props.id);
			this.props.deleteList(this.props.id);
		} catch (err) {
			console.log(err);
		}
	};

	render() {
		const { title, notes } = this.props;

		const incompleteNotes = notes
			.filter(n => !n.complete)
			.map(note => <ListItem key={note.id} {...note} complete={false} />);
		const completeNotes = notes
			.filter(n => n.complete)
			.map(note => <ListItem key={note.id} {...note} complete={true} />);

		return (
			<article className="ListCard">
				<h3>{title}</h3>
				<ul>{incompleteNotes}</ul>
				<ul>{completeNotes}</ul>
				<button className="btn" onClick={this.deleteList}>
					Delete
				</button>
				<Link to={`/notes/${this.props.id}`} key={this.props.id}>
					<button className="btn">Edit</button>
				</Link>
			</article>
		);
	}
}

export const mapDispatchToProps = dispatch => ({
	deleteList: id => dispatch(actions.deleteList(id))
});

export default connect(null, mapDispatchToProps)(ListCard);
