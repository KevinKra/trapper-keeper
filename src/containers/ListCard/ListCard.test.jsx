import React from 'react';
import { ListCard } from './ListCard';
import { mockList, mockPartialCompleteList } from '../../util/mockData';
import { shallow } from 'enzyme';

describe('ListCard', () => {
	let wrapper, mockDeleteList;
	beforeEach(() => {
		mockDeleteList = jest.fn();
		wrapper = shallow(<ListCard {...mockList} deleteList={mockDeleteList} />);
	});

	it('should match the snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('should match the snapshot if some notes are complete', () => {
		wrapper = shallow(<ListCard {...mockPartialCompleteList} deleteList={mockDeleteList} />);
		expect(wrapper).toMatchSnapshot();
	});

	describe('deleteList', () => {});
});
