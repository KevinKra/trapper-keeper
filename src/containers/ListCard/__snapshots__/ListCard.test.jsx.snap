// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListCard should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`ListCard should match the snapshot if some notes are complete 1`] = `ShallowWrapper {}`;
